pkg/infra/spot.go

```diff
@@ -268,9 +271,10 @@ func setSpotOnProviderSpec(platform configv1.PlatformType, params framework.Mach
 func setSpotOnAWSProviderSpec(params framework.MachineSetParams, maxPrice string) error {
 	spec := machinev1.AWSMachineProviderConfig{}
 
-	err := json.Unmarshal(params.ProviderSpec.Value.Raw, &spec)
-	if err != nil {
-		return fmt.Errorf("error unmarshalling providerspec: %v", err)
+	var err error
```

Comment:
I don't think this is needed here is it? Can we move the decl closer to where it's needed